### ISSUE TITLE
Copy and pasting parameters lead to results tuple being empty

### ISSUE THREAD

**Issue by XaelShan** [SasView/sasview#2975]
**Describe the bug**
After copy and pasting parameters (AND loading project) from one fit page to another, returns the error that 

> ERROR: Traceback (most recent call last): File "C:\Users\tns14\sasview\src\sas\qtgui\Perspectives\Fitting\FittingWidget.py", line 2109, in fitComplete res_list = result[0][0] ~~~~~~~~~^^^ IndexError: string index out of range

Specifically only with GPU option checked (even after getting all tests passed and pip install pyopencl)
*future edit note: after copy and pasting parameters, opencl option no longer passes all tests. To clarify, before copy/paste, all 77 opencl tests passed. After copy/paste, 60 tests _failed_*

**To Reproduce**
1. Load datasets
2. Load into separate fitting pages
3. Choose the same model for both pages
4. Click on copy on one page, paste in the other page
5. Click on compute for both, one will not form results
6. See error

**Expected behavior**
Should return a fit with copied parameters. Instead, returns error and fit does not complete; compute/plot also returns a graph with no fit/line (only data).

**Screenshots**
Before copy/paste parameters:
![image](https://github.com/user-attachments/assets/77fa5381-594b-4599-90c5-a706bcc6753a)

after copy/paste parameters (same effect as opening a .json saved project)
![image](https://github.com/user-attachments/assets/3c94754a-0256-44e4-81d4-437d76904088)

compute/plot results in this:
![image](https://github.com/user-attachments/assets/ac27a58b-a071-4078-9965-0fe131afea8f)

and fit itself does not return anything, instead just returning an ERROR

**SasView version (please complete the following information):**
 - Version: 6.0.0b1

**Operating system (please complete the following information):**
 - OS: Windows 10

**Additional context**
models that I know have this error are: core-shell-bicelle, core-shell-sphere.
Also note that any fits AFTER switching to gpu/opencl _but_ BEFORE copy/paste work. It is only AFTER copy/paste that the opencl tests no longer pass and fits no longer work.

**Comment by butlerpd** [comment]
Also identified in NIST CNR summer school

**Comment by rozyczko** [comment]
I am trying to reproduce this issue. 
However, I don't understand the connection of GPU tests failing and compute action failing as related.

I did:

1. Loaded two datasets (cyl_400_20 and cyl_400_40)
2. Selected `core_shell_bicelle` for both of them
3. Ran `Compute/Plot` for the first dataset
4. Copied parameters
5. Pasted parameters to the other set
6. Ran `Compute/Plot` for the second dataset

Both succeeded

Then I did
3. Selected some parameters
4. Ran `Fit`
5. Copied parameters from the fitted set
6. Pasted parameters to the other set
7. Ran fitting in the other set (with parameters pasted)

The fitting ran fine.

Also,
8. Did the GPU testing

This also ran fine.

Can you please provide the files used? Maybe the issue is with particular datasets?
Or maybe I am doing something wrong (which most likely is the case)

**Comment by rozyczko** [comment]
@XaelShan @butlerpd - can you please re-check the reproducibility of this issue? I am not able to see the failure following your description.

**Comment by rozyczko** [comment]
For the GPUs in the image attached, I am not able to reproduce the issue.
It might be possible the failure is Intel 620 specific?
Not being able to see the failure I am un-assigning myself

![gpus](https://github.com/user-attachments/assets/dd2512ae-15d2-42fc-b804-9108ac1388f2)

**Comment by krzywon** [comment]
I just tested reproducibility and also am unable to generate the error seen in this issue. I was able to see it before. I'll investigate to see if another fix may have resolved this.

@butlerpd - Please test this when you have a chance.

**Comment by krzywon** [comment]
The error from #2994 is eerliy similar to the one here. I'm wondering if this was fixed with #3079...

**Comment by butlerpd** [comment]
Having investigated this to some depth now I believe this actually convoluted three separate issues. The two related to copy paste appear to me to be fixed so I think we can close this issue. However we may want to open an new issue for the OpenCl portion.

**Comment by butlerpd** [comment]
The longer story:
The three issues I believe are:
* copy pasting with OpenCl on caused the copy pasted parameters to not work.  
  * This was reported by Xael who was probably not using the beta2 but some version of the release_6.0 branch? It also may have to do with many of their models (60?) failing the OpenCl tests (see below - though I think that is unlikely)
  * I was not able to reproduce the copy paste failures either with the official beta2 or the latest release_6.0.0 installer following the exact steps described above.
* The OpenCl testing issue blamed on the issue reported above.
  * I noted two problems with the OpenCl testing which are not related to the above issue.
    * when switching from No Open Cl to using Open Cl without hitting `Apply` (or `OK` but that closes the window), the tests clearly are still running with No OpenCl. However, the report at the end suggests that it DID use the GPUs. I suspect this is what confused Xael?  If you click `Apply` after selecting a new option such as "use GPU" or if you click `OK` and then open the window again, the tests do now run with the GPUs.  This is a bug that should be fixed. The report should not suggest the GPUs have been tested when they have not been
    * In my case, the GPU tests ran extremely slowly and then eventually hang on the 69th model or "superball" model.
      * After hanging, it is possible to cancel and exit the window and even to quit (with the popup "do you really want to quit") however when I say yes please quit, SasView GUI becomes unresponsive and must be killed from the Task Manager. If the model fails to pass the test it should not "get lost" and/or cause SasView to become unstable so I think this should be reported as a bug.
      * The very slow testing in general seems to be a need to recompile from source as evidenced from the log:
```python
11:46:08 - INFO: building stickyhardsphere-float64-A4EC9491 for OpenCL Intel(R) UHD Graphics 630
11:46:09 - INFO: build program: kernel 'stickyhardsphere_Iq' was part of a lengthy source build resulting from a binary cache miss (0.83 s)
11:46:09 - INFO: building superball-float32-92FDB73D for OpenCL Intel(R) UHD Graphics 630
11:47:22 - INFO: build program: kernel 'superball_Iq' was part of a lengthy source build resulting from a binary cache miss (72.45 s)
```
* Finally, the summer school copy paste report it turns out was by Brayden and unfortunately well after the fact. It came from the USANS group working on the SEEDGel datat.  I did not originally copy his report because in it he was unable to reproduce but believed it was linked to this issue. Thus I copy it below now:

> "The model in the clipboard is not the same as the currently loaded model. Not all parameters saved may paste correctly" displaying when trying to copy-paste select parameters to a different model type, even though only some parameters were selected. Could be intentional but did not seem user-intuitive. After pasting, N/A appears in the ‘value’ column of a parameter heading (the ones in bold) that are supposed to be left blank. Could not replicate today when I tried again, but I suspect that the issue is related to #2975 and would be fixed if #2975 is resolved.

In testing beta2 (released installer version) using apoferritin and shell contrast emulsion data sets and initially a core shell for both (later making the one being copied to a core shell cylinder to test when models were different) I notices that everything copied perfectly, including polydispersity and the appropriate parameters being fit checked,  **EXCEPT** for the scale parameter.  Even if the fitpage being copied to did not have polydispersity turned on, pasting the parameters that included polydispersity turned on the polydispersity in the new model as well.  The scale parameter very consistently did not copy either the value or the checked status. I was never able to see the NA value that Brayden reported however.

Doing the same tests with the latest installer from the release_6.0.0 branch now reliably **does** copy the scale parameter as well as everything else. Thus I suggest the issue(s) is/are likely fixed

**Comment by smk78** [comment]
There is another copy/paste bug I have seen in some actions builds I have been playing with of late, which is that if you have a model with polydispersity and the dispersity function is anything other than Gaussian, when you copy the parameters and then paste them into the new FitPage, the dispersity function is not pasted and the model resets to the Gaussian. This behaviour is, of course, non-obvious to the User.

**Comment by krzywon** [comment]
Okay, here's a new error:

- Run a fit with any dataset/model/GPU option.
- Change the GPU option in the GPU window and click OK or Accept.
- Attempt to run the fit from the same fit window.
- Error thrown.

It looks like the sasmodels environment reset is not working properly at this point. This is new since v5. I won't hold up the RC for it, but might try to fix it for the final release.

**Comment by krzywon** [comment]
> Okay, here's a new error:
> 
> * Run a fit with any dataset/model/GPU option.
> * Change the GPU option in the GPU window and click OK or Accept.
> * Attempt to run the fit from the same fit window.
> * Error thrown.
> 
> It looks like the sasmodels environment reset is not working properly at this point. This is new since v5. I won't hold up the RC for it, but might try to fix it for the final release.

Note - this only fails when the original GPU option was anything but `None`. Switching from one OpenCL/Cuda option to any other option results in the errors. This behavior is the same from v5.0.6 to now so this will not be fixed for v6.0.
